src/visitAllChildren/types.ts

```diff
@@ -0,0 +1,20 @@
+import ts from 'typescript';
+
+export type VisitorFunction<N extends ts.Node = ts.Node> = (node: N) => unknown;
+
+export type VisitorsMap = VisitorsWithSpecificNodeTypes &
+  VisitorsWithGenericNodeType &
+  VisitorsAllNodes;
+
+type VisitorsAllNodes = {
+  all?: VisitorFunction;
+};
+
+type VisitorsWithGenericNodeType = {
+  [k in Exclude<ts.SyntaxKind, keyof VisitorsWithSpecificNodeTypes>]?: VisitorFunction;
+};
+
+type VisitorsWithSpecificNodeTypes = {
+  [ts.SyntaxKind.InterfaceDeclaration]?: VisitorFunction<ts.InterfaceDeclaration>;
```

Comment:
Specific node types per visitor should be added here. If the specific match is not implemented, the visitor receives the default type `ts.Node`.

New entries here automatically get excluded from type `VisitorsWithGenericNodeType`